MindscapeHQ/raygun4android
# Build modernisation to modern Gradle/Android Studio

## Summary

### What problem(s) does this PR solve?

Fixes the AS Gradle builds in so far that it's now up to date again with current Android studio version.

### Type of change

- [x] Internal to project

### What was done to solve the problem?

- Updated Gradle and AGP versions progressively and fixed issues arising along the way
- Replaced KAE (deprecated now) with view binding in sample app
- Set version to 4.1-SNAPSHOT for the time being (more changes to come)

## QA

- Project should be able to import in current stable AS again and sync/build
- :app and :provider modules (the latter via the app build) build correctly when running the sample app in an Emulator (tested with SDK 33 for now)
